torch/csrc/autograd/functions/utils.h

```diff
@@ -35,6 +35,11 @@ struct ComputeRequiresGrad : IterArgs<ComputeRequiresGrad> {
     if (var.defined() && var.requires_grad()) {
       out = true;
     }
+    TORCH_CHECK(!tensor.defined() || !tensor.is_view()
+      || static_cast<DifferentiableViewMeta*>(tensor.unsafeGetTensorImpl()->autograd_meta())->get_creation_meta() != CreationMeta::NO_VARIABLE_TYPE_VIEW,
+      "Input contains a view created in InferenceMode without proper grad_fn setup, "
```

Comment:
I am not sure about these error messages.
Do you have an example of user code that would raise this?

Reply:
Yup https://github.com/pytorch/pytorch/pull/53343/files#diff-78dfc27f4cd89424e2e342c82969d194f104d138ac70eb1069192735068af21eR260

Reply:
I would phrase this slightly differently to be more consistent with this kind of errors in gradmode (https://github.com/pytorch/pytorch/blob/76129c7cdfee6f33861a8b1602326a98bd6c22d0/torch/csrc/autograd/variable.cpp#L510)

This error only happens when GradMode is enabled right?
This error happens for any out of place op? If so, then calling `.clone()` would also throw and error? Do we have a better suggestion?

Reply:
I replaced it with detach(), I feel this one is a better suggestion?